### DISCUSSION TITLE
Different semantic commit message depending on where the package got updated

### DISCUSSION THREAD

**Discussion by kvnwolf** [renovatebot/renovate#17729] [Request Help]
Let's say I have a monorepo with the following structure:

```
.
├── .github
│   └── workflows
│       ├── release.yml
│       └── validate.yml
├── package.json
└── packages
    ├── a
    │   └── package.json
    └── b
        └── package.json
```

What I want to achieve, is having different semantic commit messages depending on _where_ a dependency was updated, in order to generate better changelogs.

For instance:

- For a dependency updated in `.github/workflows/release.yml`, I want the commit message: `ci: update dependency X to vX.X.X`
- For a dependency updated in `.github/workflows/validate.yml`, I want the commit message: `ci: update dependency X to vX.X.X`
- For a dependency updated in `package.json`, I want the commit message: `chore: update dependency X to vX.X.X`
- For a dependency updated in `packages/a/package.json`, I want the commit message: `deps(@my-lib/a): update dependency X to vX.X.X`
- For a dependency updated in `packages/b/package.json`, I want the commit message: `deps(@my-lib/b): update dependency X to vX.X.X`

**Comment by viceice** [comment]
you need package rule for each folder, match the paths and change branch name and other appropriate settings for commit message

**Reply by kvnwolf** [reply]
Thanks for your answer! Do you have an example on how to to this? I am pretty new to using Renovate.

**Reply by viceice** [reply]
https://docs.renovatebot.com/configuration-options/#packagerules
https://docs.renovatebot.com/configuration-options/#semanticcommittype
https://docs.renovatebot.com/configuration-options/#additionalbranchprefix
